db_autorestore.md

```diff
@@ -0,0 +1,89 @@
+ * documentdb_latest_snapshot_id
+ * main_cicd_name
+
+The current value will be the one from which the databases are restored from and the latest value will be the value of the most recent snapshot that exists in the account for DocumentDB or Aurora.
+We need to have 2 different parameters, one for the current and one for the latest snapshot, to be able to restore the databases from a specific snapshot ID outside of this process.
+There are 2 pipeline in the account, so the name of the main pipeline, the one that is deploying the infrastructure, was needed.
+
+The entire solution was written using IaC in AWS CDK with Python.
+
+![Architecture](db_autorestore)
+
+## How does it work
+Using a cronjob the update function (*db-update-latest-snapshot-id*) will run every 2 weeks at 2 AM.
+This function has the following tasks/functions:
+
+* *copy_docdb_snapshot* - checks for the latest DocumentDB shared and manual snapshot, and it takes the latest value for both of them. It then checks if the latest shared snapshot is already copied and if not, it will copy that one, if yes, it will return a nice message. 
+
+* *get_latest_rds_snapshots* - checks for the latest shared snapshot for Aurora DB and it gets the latest one.
+
+* *update_ssm* - it will update the SSM Parameter store with the new values of the latest snapshots in *documentdb_latest_snapshot_id* and in *aurora_latest_snapshot_id*.
+
+And this is it for the update process.
+Now for the restore is even simpler. Using another cronjob the restore function  (*db-restore-from-snapshot*) will run every 2 weeks, 2 hours later than the update process.
+The tasks of this functions are:
+
+* *update_current* - it gets the value from the latest parameters and copies it to the current one. Now the value of the *documentdb_current_snapshot_id* will be updated with the value of *documentdb_latest_snapshot_id*. The pipeline will check the value of the current parameters. We have split this process with latest and current parameters to facilitate the restore from a snapshot in other cases than the ones described in this article as menioned above.
+
+* trigger the deployment pipeline using the *main_cicd_name* from the SSM Parameter Store
+
+    ```
+    # Trigger a new pipeline deployment
+        cicd_client.start_pipeline_execution(
+            name=cicd_name
+        )
+    ```
+
+## Small bumps
+The process of finding out a solution was challenging in each phase of implementation. At the beginning we tried to find a solution that will come on top of the existing one and work to improve it. In the process of implementing the solution with Lambda functions there was the problem of identifying the exact snapshot we need, which of course was just a matter of knowing what you want to extract and reading the documentation. For example, the process of checking the available shared snapshot for the DocumentDB is looking like this:
+
+```
+docdb_shared_snapshots = docdb_client.describe_db_cluster_snapshots(
```

Comment:
Shouldn't you use a paginator here? Or will there never be more then a few snapshots?